pkg/providers/oidc/verifier.go

```diff
@@ -107,3 +120,30 @@ func (v *idTokenVerifier) interfaceSliceToString(slice interface{}) []string {
 	}
 	return strings
 }
+
+func (v *idTokenVerifier) verifyEmail(token *oidc.IDToken) (bool, error) {
+
+	if v.verificationOptions.AllowUnverifiedEmail {
+		return true, nil
+	}
+
+	var claims struct {
+		Subject  string `json:"sub"`
+		Email    string `json:"email"`
+		Verified *bool  `json:"email_verified"`
+	}
+
+	if err := token.Claims(&claims); err != nil {
+		return false, fmt.Errorf("failed to parse bearer token claims: %v", err)
+	}
+
+	if claims.Email == "" {
+		claims.Email = claims.Subject
+	}
+
+	if claims.Verified != nil && !*claims.Verified {
+		return false, fmt.Errorf("email in id_token (%s) isn't verified", claims.Email)
+	}
```

Comment:
I see you added the code here but it seems like you missed to remove the same code from `CreateTokenToSessionFunc` in `session.go`. Therefore the old check is still running over there once again over your new implementation.